### PR TITLE
Remove deprecated methods

### LINKED ISSUE #281: Remove deprecated get array elements methods
As part of this issue, remove the deprecated get/release/commit_<type>_array_{elements|critical} methods.

At the same time, remove _auto_ from the remaining array elements/critical methods. That is, follow/match the underlying JNI method names in these methods.

### PR DESCRIPTION
**Pull request by maurolacy**
## Overview

Closes #281.

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by [automated tests](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/jni-rs/jni-rs/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has documentation
- [x] User-visible changes are mentioned in the Changelog
- [ ] The continuous integration build passes

### CODE REVIEW

File: tests/jni_api.rs [hunk 1]
```diff
@@ -517,53 +476,6 @@ pub fn get_auto_long_array_elements_commit() {
...
-    // Confirm modification of original Java array
-    let mut res: [i8; 3] = [0; 3];
-    env.get_byte_array_region(java_array, 0, &mut res).unwrap();
-    assert_eq!(res[0], 2);
-    assert_eq!(res[1], 3);
-    assert_eq!(res[2], 4);
-}
-
 #[test]
 pub fn get_auto_primitive_array_critical() {
```
Comment (dmitry-timofeev) [inline tests/jni_api.rs:480]: Nit: tests could also drop "auto"

**Review by dmitry-timofeev — APPROVED** [review]
LGTM, thanks!

**Review by stanislav-tkach — APPROVED** [review]
:+1:

**Comment by stanislav-tkach** [comment]
It isn't related to your pull request, but we have some [issues](https://github.com/jni-rs/jni-rs/pull/282/checks?check_run_id=1343885451):
```
 ReleasePrimitiveArrayCritical: release array failed bounds check, incorrect pointer returned ? array: 0x00007f4de007bc40 carray: 0x00007f4dd8001260
GuardedMemory(0x00007f4de445c420) base_addr=0x00007f4dd8001240 tag=0x00007f4dd8001720 user_size=24 user_data=0x00007f4dd8001260
  Header guard @0x00007f4dd8001240 is BROKEN
  Trailer guard @0x00007f4dd8001278 is OK
  User data appears to have been freed
FATAL ERROR in native method: ReleasePrimitiveArrayCritical: failed bounds check
error: test failed, to rerun pass '--test jni_api'
Error: test failed, to rerun pass '--test jni_api'
Caused by:
  process didn't exit successfully: `/home/runner/work/jni-rs/jni-rs/target/debug/deps/jni_api-74ed8f7f93a96c88` (signal: 6, SIGABRT: process abort signal)
```

**Comment by maurolacy** [comment]
> It isn't related to your pull request, but we have some [issues](https://github.com/jni-rs/jni-rs/pull/282/checks?check_run_id=1343885451):
> 
> ```
>  ReleasePrimitiveArrayCritical: release array failed bounds check, incorrect pointer returned ? array: 0x00007f4de007bc40 carray: 0x00007f4dd8001260
> GuardedMemory(0x00007f4de445c420) base_addr=0x00007f4dd8001240 tag=0x00007f4dd8001720 user_size=24 user_data=0x00007f4dd8001260
>   Header guard @0x00007f4dd8001240 is BROKEN
>   Trailer guard @0x00007f4dd8001278 is OK
>   User data appears to have been freed
> FATAL ERROR in native method: ReleasePrimitiveArrayCritical: failed bounds check
> error: test failed, to rerun pass '--test jni_api'
> Error: test failed, to rerun pass '--test jni_api'
> Caused by:
>   process didn't exit successfully: `/home/runner/work/jni-rs/jni-rs/target/debug/deps/jni_api-74ed8f7f93a96c88` (signal: 6, SIGABRT: process abort signal)
> ```

Yes, I've seen it locally, but only on one machine... was investigating it before asking about it / creating an issue.

This happens when `commit()` is called. Not sure exactly when / on which platform, but doesn't seem related to the Rust version. Maybe to the JNI / java version. I'm still investigating.

**Comment by maurolacy** [comment]
See #283.

**Pull request merged**